scripts/zones/Ifrits_Cauldron/mobs/Bomb_Queen.lua

```diff
@@ -16,23 +16,57 @@ entity.onMobInitialize = function(mob)
 end
 
 entity.onMobSpawn = function(mob)
-    mob:setLocalVar("petCooldown", os.time() + 5) -- five seconds for first pet
+    mob:setLocalVar("spawn_time", os.time() + 5) -- five seconds for first pet
 end
 
 entity.onMobFight = function(mob, target)
-    -- every ~20 seconds after first pet, a new pet will spawn around queen's position.
-    -- 49.5% (198/400) chance for Prince/Princess. 1% (4/400) chance for Bomb Bastard.
-    if os.time() > mob:getLocalVar("petCooldown") then
-        local petId = mob:getID() + 1 + math.floor(math.random(0, 399) / 99)
-        local pet = GetMobByID(petId)
-
-        if pet and not pet:isSpawned() then
-            local pos = mob:getPos()
-            pet:setSpawn(pos.x + math.random(-2, 2), pos.y, pos.z + math.random(-2, 2), pos.rot)
-            pet:spawn()
-            pet:updateEnmity(target)
-
-            mob:setLocalVar("petCooldown", os.time() + 20)
+    if os.time() >= mob:getLocalVar("spawn_time") then
+        mob:setLocalVar("spawn_time", os.time() + 30)
+        local mobId = mob:getID()
+        local canSpawnPet = false
+        for id = mobId + 1, mobId + 5 do
+            if GetMobByID(id):getCurrentAction() == 0 then
```

Comment:
getCurrentAction was slated to be deprecated at one point, and I'm not sure of its future, but if we're gonna use it lets use the enums: `xi.action.NONE` instead of 0